Libplanet/PoS/Delegation.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Assets;
+
+namespace Libplanet.PoS
+{
+    internal class Delegation : DelegationInfo
+    {
+        internal Delegation(Address delegatorAddress, Address validatorAddress)
+            : base(delegatorAddress, validatorAddress)
+        {
+        }
+
+        internal Delegation(IValue serialized)
+            : base(serialized)
+        {
+        }
+
+        internal Delegation(DelegationInfo delegationInfo)
+            : base(delegationInfo)
+        {
+        }
+
+        internal static (IAccountStateDelta, FungibleAssetValue) Bond(
+            IAccountStateDelta states,
+            FungibleAssetValue consensusToken,
```

Comment:
IMO this method is not good at here. I suggest making a new file for this method.
Because `Bond` is not only used `Delegation.Delegate`, but also `Redelegaion.Redelegate`.
If you think `Delegation` is an inner concept about `Redelegation`, The design needs to be changed to better reflect that concept.

Reply:
I agree that this structure doesn't seem so good.
But I don't think `Delegation` is an inner concept about `Redelegation`.
`Redelegation` uses methods `Delegation.Bond` and `Undelegation.Unbond`, so it's between those.
I think there are some better structure, but let me think about it more.